File: 01_Language/01_Functions/js/array_replace.js
function array_replace(arr) {
    let res = {}
    for (let i = 0; i < arguments.length; i++) {
        for (let j in arguments[i]) {
            res[j] = arguments[i][j]
        }
    }
    return res
}

let base = ["orange", "banana", "apple", "raspberry"]
let replacements = {0: "pineapple", 4: "cherry"}
let replacements2 = {0: "grape"}

console.log(array_replace(base, replacements, replacements2));

File: 01_Language/01_Functions/python/array_replace.py
# coding: utf-8

def array_replace(*args):
    d = {}
    for i in args:
        if isinstance(i, dict):
            for k, v in i.items():
                d[k] = v
        else:
            for k, v in enumerate(i):
                d[k] = v
    return d


if __name__ == '__main__':
    base = ["orange", "banana", "apple", "raspberry"]
    replacements = {0: "pineapple", 4: "cherry"}
    replacements2 = {0: "grape"}

    print(array_replace(base, replacements, replacements2))
